src/compiler/checker.ts

```diff
@@ -6662,46 +6677,95 @@ namespace ts {
             return s.valueDeclaration ? s.valueDeclaration.kind : SyntaxKind.PropertyDeclaration;
         }
 
-        function getDeclarationFlagsFromSymbol(s: Symbol) {
+        function getDeclarationFlagsFromSymbol(s: Symbol): NodeFlags {
             return s.valueDeclaration ? getCombinedNodeFlags(s.valueDeclaration) : s.flags & SymbolFlags.Prototype ? NodeFlags.Public | NodeFlags.Static : 0;
         }
 
-        function checkClassPropertyAccess(node: PropertyAccessExpression | QualifiedName, left: Expression | QualifiedName, type: Type, prop: Symbol) {
+        /**
+         * Check whether the requested property access is valid.
+         * Returns true if node passed the check, and false otherwise.
+         * @param node The node to be checked.
+         * @param left The left hand side of the property access (eg: the super is `super.*`).
+         * @param type The type of left.
+         * @param prop The symbol for the right hand side of the property access.
+         */
+        function checkClassPropertyAccess(node: PropertyAccessExpression | QualifiedName, left: Expression | QualifiedName, type: Type, prop: Symbol) : boolean {
             let flags = getDeclarationFlagsFromSymbol(prop);
-            // Public properties are always accessible
+            let declaringClass : InterfaceType;
+
+            if (left.kind === SyntaxKind.SuperKeyword) {
+                let errorNode = (<PropertyAccessExpression>node).name ?

```

Comment:
Perhaps what you should be checking is whether or not `node.kind` is a property access